doc/reference/networking/net_l2.rst

```diff
@@ -64,85 +65,72 @@ There are, however, two differences:
 - the driver_api pointer must point to a valid :c:type:`struct
   net_if_api` pointer.
 
-- The network device driver must use ``NET_DEVICE_INIT_INSTANCE()``. This
-  macro will call the DEVICE_AND_API_INIT() macro, and also
+- The network device driver must use ``NET_DEVICE_INIT_INSTANCE()``
+  or ``ETH_NET_DEVICE_INIT()`` for Ethernet devices. These
+  macros will call the ``DEVICE_AND_API_INIT()`` macro, and also
   instantiate a unique :c:type:`struct net_if` related to the created
   device driver instance.
 
 Implementing a network device driver depends on the L2 stack it
-belongs to: Ethernet, IEEE 802.15.4, etc. In the next section, we will
-describe how a device driver should behave when receiving or sending a
-packet. The rest is really hardware dependent and thus does not need
-to be detailed here.
+belongs to: :ref:`Ethernet <ethernet_interface>`,
+:ref:`IEEE 802.15.4 <ieee802154_interface>`, etc.
+In the next section, we will describe how a device driver should behave when
+receiving or sending a network packet. The rest is really hardware dependent
```

Comment:
Suggested:

"The rest is really hardware dependent" => "The rest is hardware dependent"

"and thus does not need to be detailed here." => "and is not detailed here."

Because who known, maybe it's not "really", but "somewhat" hardware dependent. And may be it should be detailed here (more details and examples are always good), but we don't, just don't.